README.md

```diff
@@ -29,99 +31,99 @@ Tested in Node 4+.
 
 ## Usage
 
+Create a cosmiconfig explorer, then either `search` for or directly `load` a configuration file.
+
 ```js
-var cosmiconfig = require('cosmiconfig');
-
-var explorer = cosmiconfig(yourModuleName[, options]);
-
-explorer.search()
-  .then((result) => {
-    // result.config is the parsed configuration object
-    // result.filepath is the path to the config file that was found
-  })
-  .catch((parsingError) => {
-    // do something constructive
-  });
+const cosmiconfig = require('cosmiconfig');
+// ...
+const explorer = cosmiconfig(moduleName);
+
+// Search for a configuration by walking up directories.
+// See documentation for search, below.
+explorer.search().then(
+  (result) => {
+    // result.config is the parsed configuration object.
+    // result.filepath is the path to the config file that was found.
+    // result.isEmpty is true if there was nothing to parse in the config file.
+  },
+  (parsingError) => {
```

Comment:
I prefer the `.catch` equivalent as it seems clearer and explicit(to me at least).

Reply:
It's less precise, though: if your callback that does something with `result` throws an error, that error is going to end up in the `catch` statement but will not be a parsing error.

Reply:
I'll change it to `catch` and just change the `parsingError` name to `error`.